### PR TITLE
factorize capacity computations

### PR DESCRIPTION
**Pull request by petchema**
Factorize capacity computations (player carry capacity, wagon hold
capacity, gold dropping into wagon), using floats to avoid rounding down
bugs.
Code that makes horse, wagon and arrows weightless was also factorized.

Gold dropping message was left as-is to preserve Daggerfall classic
behavior; However, when dropping gold into wagon, a post capacity check
was added: if the wagon cannot accomodate the amount, only the gold that
fits is dropped and a message is displayed.

Bug report https://forums.dfworkshop.net/viewtopic.php?f=24&t=1587

I found another bug while factorizing: gold weight was counted twice in
wagon, because it was handled with a special case when the common case
was sufficient.

### CODE REVIEW

File: Assets/Scripts/Game/UserInterfaceWindows/DaggerfallInventoryWindow.cs [hunk 1]
```diff
@@ -1156,9 +1157,19 @@ private void DropGoldPopup_OnGotUserInput(DaggerfallInputMessageBox sender, stri
...
+            {
+                // Check wagon weight limit
+                int wagonCanHold = ComputeCanHoldAmount(playerGold, DaggerfallBankManager.goldUnitWeightInKg, ItemHelper.wagonKgLimit - remoteItems.GetWeight());
+                if (goldToDrop > wagonCanHold)
+                {
+                    goldToDrop = wagonCanHold;
+                    DaggerfallMessageBox messageBox = new DaggerfallMessageBox(uiManager, this);
+                    messageBox.SetText(String.Format(wagonFullGold, wagonCanHold));
+                    messageBox.ClickAnywhereToClose = true;
+                    messageBox.Show();
```
Comment (ajrb) [inline Assets/Scripts/Game/UserInterfaceWindows/DaggerfallInventoryWindow.cs]: Maybe you could use this to shorten this boilerplate code for all the MsgBoxes you added:

DaggerfallUI.MessageBox(text);

**Comment by ajrb** [comment]
Excellent work Pango, looks good.

**Comment by ajrb** [comment]
Perfect, very happy how you sorted this out.

**Comment by Interkarma** [comment]
Very clever! Thank you. :)

**Pull request merged**